### PR TITLE
Adding ML Tutorial

### PR DESCRIPTION
**Pull request by rahul799**

### CODE REVIEW

File: docs/source/ML_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+.. code-block:: python
+
+    xg_reg.fit(X_train, Y_train)
+    Y_pred = xg_reg.predict(X_test)
+
+Compute the mean squared error (mse) by invoking the mean_sqaured_error function from sklearn's metrics module.
+
+.. code-block:: python
+
+    mse = np.sqrt(mean_squared_error(Y_test, Y_pred))
```
Comment (timosachsenberg) [inline docs/source/ML_tutorial.rst:183]: is this the mean absolute error?

Comment (jpfeuffer) [reply docs/source/ML_tutorial.rst:183]: No. It is the root mean squared error.

Comment (timosachsenberg) [reply docs/source/ML_tutorial.rst:183]: ah I think we had a miscommunication in the meeting. I will change that back.

**Comment by jpfeuffer** [comment]
File links need to be updated. Currently the consensus seems to be to always use urlretrieve (which I still don't like though).

And an entry in the TOC is missing

**Comment by rahul799** [comment]
@jpfeuffer I didn't understand what is TOC :confused:

**Comment by jpfeuffer** [comment]
Table of contents. The sidebar.

**Comment by rahul799** [comment]
ohk !!! It's fixed now.

**Pull request merged**